redpanda/templates/statefulset.yaml

```diff
@@ -57,6 +57,12 @@ spec:
       securityContext:
 {{- toYaml .Values.statefulset.podSecurityContext | nindent 8 }}
       initContainers:
+        - name: set-datadir-ownership
+          image: busybox:latest
+          command: ["/bin/sh", "-c", "chown 101:101 -R /var/lib/redpanda/data"]
+          volumeMounts:
+            - name: datadir
+              mountPath: /var/lib/redpanda/data
```

Comment:
Is there a case where the user might want to change the default data_directory (/var/lib/redpanda/data)?

Reply:
I think changing the default directory in a kubernetes (or helm) situation would be much less likely (than local install) since this is the container path, and this path is just a mount point for a persistent volume which could be multiple places (a completely different directory on the node, a remote storage device through some persistent storage cloud service, etc.). I can't think of a scenario right now where the customer would be using this path, unless they are connecting directly into the pod for some reason.

At the same time, this would be easy to provide as a modifiable parameter in `values.yaml` if we can think of a good reason to do so.

Reply:
Thanks! It's clear to me now :)